### DISCUSSION TITLE
AttributeError in custom command

### DISCUSSION THREAD

**Discussion by brush701** [saulpw/visidata#2490] [Q&A] [answered]
I'm trying to add a custom command that computes the frequency of numerical columns. I've got it mostly working, except that every time I invoke it I get a status warning: `AttributeError: 'int' object has no attribute 'name'`. Other than that the command executes correctly, including setting the column name. Any suggestions on where I can fix this error.

```
Sheet.addCommand('z.', 'addcol-freq', 'addColumn(FrequencyColumn(cursorCol), cursorColIndex+1)')

def FrequencyColumn(sourceCol):
    return Column(sourceCol.name+"_frequency",
                    getter=lambda c,r: c.source.getTypedValue(r)/c.total,
                    source=sourceCol,
                    total=sum(sourceCol.getValues(sourceCol.sheet.rows)))
```

**Comment by midichef** [comment, marked as answer]
Every argument to `addColumn()` except the named argument `index` is treated like a list of columns because of `*cols`:
https://github.com/saulpw/visidata/blob/8f823f5a3d922756e49ed47086419ab366e8fbc4/visidata/sheets.py#L520

The solution is to use `index=` in front of your column index:

`Sheet.addCommand('z.', 'addcol-freq', 'addColumn(FrequencyColumn(cursorCol), index=cursorColIndex+1)')`

Also, a feature that may help you as you write code for visidata: to examine a traceback for errors like this, run the command `error-recent` (with the shortcut `Ctrl+E`). Or if you have more than one error, `errors-all`.

In this example, the traceback would offer this clue on where to look or add debug statements:
```
  File "/home/midichef/.local/lib/python3.10/site-packages/visidata/sheets.py", line 535, in addColumn
    col.name = self.maybeClean(col.name)
AttributeError: 'int' object has no attribute 'name'
```

**Reply by brush701** [reply]
Thanks, this was it exactly!
